src/features/CourtCaseFilters/CourtCaseFilterWrapper.tsx

```diff
@@ -16,44 +17,66 @@ const CourtCaseFilterWrapper: React.FC<Props> = ({
   paginationTop,
   paginationBottom
 }: Props) => {
-  const [areAppliedFiltersShown, setAreAppliedFiltersShown] = useState(false)
+  const user = useCurrentUser()
+  const filterPanelKey = `is-filter-panel-visible-${user.username}`
+  const [isSearchPanelShown, setIsSearchPanelShown] = useState(true)
+  console.log(new Date())
+
+  useEffect(() => {
+    const filterPanelValue = localStorage.getItem(filterPanelKey)
+    if (filterPanelValue) {
+      const storedDate = new Date(filterPanelValue)
+      storedDate.setDate(storedDate.getDate() + 7)
+      if (storedDate > new Date()) {
+        setIsSearchPanelShown(false)
+      } else {
+        localStorage.removeItem(filterPanelKey)
+        setIsSearchPanelShown(true)
+      }
+    }
+  }, [filterPanelKey])
+
   return (
     <>
-      <div className={`moj-filter-layout`}>
-        <div className="moj-filter-layout__filter">
-          <div className={areAppliedFiltersShown ? "moj-filter" : "moj-filter moj-hidden"}>{filter}</div>
-        </div>
-        <Heading className="hidden-header" as="h1" size="LARGE">
-          {"Case list"}
-        </Heading>
-        <div className="moj-filter-layout__content">
-          <div className="moj-button-menu">
-            <div className="moj-action-bar">
-              <button
-                data-module="govuk-button"
-                id="filter-button"
-                className="govuk-button govuk-button--secondary govuk-!-margin-bottom-0"
-                type="button"
-                aria-haspopup="true"
-                aria-expanded="false"
-                onClick={() => {
-                  setAreAppliedFiltersShown(!areAppliedFiltersShown)
-                }}
-              >
-                {areAppliedFiltersShown ? "Hide search panel" : "Show search panel"}
-              </button>
-              {!areAppliedFiltersShown && <div className="moj-button-menu__wrapper">{appliedFilters}</div>}
-            </div>
+      <div className="moj-filter-layout__filter">
+        <div className={isSearchPanelShown ? "moj-filter" : "moj-filter moj-hidden"}>{filter}</div>
+      </div>
+      <Heading className="hidden-header" as="h1" size="LARGE">
+        {"Case list"}
+      </Heading>
+      <div className="moj-filter-layout__content">
+        <div className="moj-button-menu">
+          <div className="moj-action-bar">
+            <button
+              data-module="govuk-button"
+              id="filter-button"
+              className="govuk-button govuk-button--secondary govuk-!-margin-bottom-0"
+              type="button"
+              aria-haspopup="true"
+              aria-expanded={isSearchPanelShown === true}
+              onClick={() => {
+                const newValue = !isSearchPanelShown
```

Comment:
`newValue` here should probably be better named, maybe `showSearchPanel`?

Reply:
done